File: renamer/renamer.rb
#!/usr/bin/env ruby

require 'pathname';

class Renamer
	def initialize()
		@path = Pathname.new(ARGV[0]).realpath;
		@prefix = ARGV[1];
		@extensions = ['jpg', 'png' , 'gif'];
		@all_files = Array.new;
		@images = Array.new;
		Dir.foreach(@path) { |file|
			@all_files << file;
		}		
	end;

	def getImages() 
		@all_files.each() { |file|
			ext = file.split(/([\w\d]{3}$)/)[1];
			@extensions.each() {
				|extension|
				if (ext == extension) 
					@images << file;
					self.rename(file);					
				end;
			}	
		}
	end;

	def rename(file) 
		File.rename(File.join(@path, file), File.join(@path, @prefix + file));			
	end;
end;


def main() 
	rename = Renamer.new;
	rename.getImages();
end;

if (__FILE__ == $0) 
	main();
end;                                     

File: renamer/renamer.py
#!/usr/bin/env python

import os;
from sys import argv;

class renamer:
	def __init__(self):
		try:
			self.path = os.path.abspath(argv[1]);
		except:
			exit(1);
		try:
			self.prefix = argv[2];
		except:
			self.prefix = argv[1];
		self.filenames = os.listdir(argv[1]);
		self.extensions = ['jpg', 'gif', 'png'];
		self.files = [];

	def getImages(self):
		for file in self.filenames:
			if (file.rfind('.') != -1):
				ext = file[(file.rfind('.')+1):].lower();
				for extension in self.extensions:
					if (ext == extension.lower() and os.path.isfile(os.path.join(self.path, file))):
						self.files.append(file);
						self.rename(file);

	def rename(self, file):
		image_path = os.path.join(self.path, file);
		os.rename(image_path, os.path.join(self.path, self.prefix + file));
	

def main():
	rename = renamer();
	rename.getImages();


if (__name__ == "__main__"):
	main();	
